src/components/meetups/MeetupListPage.tsx

```diff
@@ -31,10 +39,54 @@ const styles = StyleSheet.create({
 });
 
 export default function MeetupListPage({ navigation }: MeetupListPageProps) {
-  const [pendingMeetups] = useState(["ABC", "DEF", "ABC", "DEF"]);
-  const [upcomingMeetups] = useState(["GHI", "JKL", "GHI", "JKL"]);
-  const [unratedMeetups] = useState(["MNO", "PQR", "MNO", "PQR"]);
+  // Define the core sections of the list
+  const [pendingMeetups, setPendingMeetups] = useState<string[]>([]);
+  const [upcomingMeetups, setUpcomingMeetups] = useState<string[]>([]);
+  const [unratedMeetups, setUnratedMeetups] = useState<string[]>([]);
+
+  // Create state for loading indication
+  const [isLoading, setLoading] = useState(false);
 
+  // Retrieve account information
+  const { refreshToken } = useTokenState();
+  const {
+    account: { email },
+  } = useAccountState();
+
+  // Create function to load and set all data
+  const loadMeetupData = async () => {
+    setLoading(true);
+    // Load the required sections
+    const [pendingMeetupList] = await getPendingMeetupsList(
+      refreshToken,
+      email
+    );
+    const [upcomingMeetupList] = await getUpcomingMeetupsList(
+      refreshToken,
+      email
+    );
+    const [unratedMeetupList] = await getUnratedMeetupsList(
+      refreshToken,
+      email
+    );
+    // Set the sections once loaded
+    setPendingMeetups(pendingMeetupList.map((meetup) => meetup.id));
+    setUpcomingMeetups(upcomingMeetupList.map((meetup) => meetup.id));
+    setUnratedMeetups(unratedMeetupList.map((meetup) => meetup.meetup_id));
+    setLoading(false);
+  };
+
+  // Allows automatic loading on the page when navigating to the page from a nested or parent screen
+  useEffect(() => {
+    const unsubscribe = navigation.addListener("focus", () => {
+      loadMeetupData();
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, [navigation]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
this looks good but I think its gonna change / conflict with my PR for routing

Reply:
Sounds good, we will discuss this in the meeting then